pkg/component/component.go

```diff
@@ -68,6 +68,31 @@ func CreateFromGit(client *occlient.Client, name string, ctype string, url strin
 	return nil
 }
 
+// GetComponentPorts provides slice of ports used by the component in the form port_no/protocol
+func GetComponentPorts(client *occlient.Client, componentName string, applicationName string) (res []string, err error) {
+	componentLabels := componentlabels.GetLabels(componentName, applicationName, false)
+	componentSelector := util.ConvertLabelsToSelector(componentLabels)
+
+	dc, err := client.GetOneDeploymentConfigFromSelector(componentSelector)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to fetch deployment configs for the selector %v", componentSelector)
+	}
+	// Find Pod for component
+	podSelector := fmt.Sprintf("deploymentconfig=%s", dc.Name)
+	// Wait for Pod to be in running state otherwise we can't sync data to it.
+	pod, err := client.WaitAndGetPod(podSelector)
```

Comment:
I guess we can get the containers and thus the ports from `dc.Spec.Template.Spec.Containers` instead of waiting to retrieve the Pod as we have already retrieved the dc in the previous lines.

Reply:
Ack